### DISCUSSION TITLE
12.1.0 > 12.2.0 missing internal temp sensor

### DISCUSSION THREAD

**Discussion by Saentist** [arendst/Tasmota#17291] [Q&A] [answered]
### PROBLEM DESCRIPTION
Disappear internal temperature sensor.
Device M5Stack M5Stamp C3U Module (K122)

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): M5Stack M5Stamp C3U Module (K122)
- [x] Tasmota binary firmware version number used: 12.2.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: OTA
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
08:23:32.149 CMD: Backlog Template; Module; GPIO 255
08:23:32.171 MQT: stat/tasmota_265344/RESULT = {"NAME":"M5Stamp C3U","GPIO":[1,1,1376,1,1,1,1,1,1,32,1,0,0,0,0,0,0,0,1,1,1,1],"FLAG":0,"BASE":1}
08:23:32.401 MQT: stat/tasmota_265344/RESULT = {"Module":{"0":"M5Stamp C3U"}}
08:23:32.653 MQT: stat/tasmota_265344/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"1376":"WS28121"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"32":"Button1"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
08:29:39.773 CMD: Status 0
08:29:39.778 MQT: stat/tasmota_265344/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_265344","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
08:29:39.782 MQT: stat/tasmota_265344/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32c3.bin","RestartReason":"Software reset digital core","Uptime":"0T07:49:52","StartupUTC":"2022-12-05T23:39:47","Sleep":50,"CfgHolder":4617,"BootCount":61,"BCResetTime":"2022-04-09T13:46:27","SaveCount":390}}
08:29:39.788 MQT: stat/tasmota_265344/STATUS2 = {"StatusFWR":{"Version":"12.2.0(tasmota)","BuildDateTime":"2022-10-17T08:39:53","Core":"2_0_5","SDK":"v4.4.3.4","CpuFrequency":160,"Hardware":"ESP32-C3 rev.3","CR":"395/699"}}
08:29:39.792 MQT: stat/tasmota_265344/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["MikroTik-4EFD03",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
08:29:39.800 MQT: stat/tasmota_265344/STATUS4 = {"StatusMEM":{"ProgramSize":1479,"Free":1856,"Heap":180,"StackLowMark":2,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"164020","FlashFrequency":80,"FlashMode":"DOUT","Features":["00000809","9F9AD7CF","0015A001","B7F7BFCF","05DA9BC0","E0360DC7","000840D2","20200000","D434482D","00000001"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,14,16,17,20,21,24,26,27,29,34,35,38,50,52,59,62,63,86,87","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,28,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,98,127"}}
08:29:39.806 MQT: stat/tasmota_265344/STATUS5 = {"StatusNET":{"Hostname":"tasmota-265344-4932","IPAddress":"192.168.1.92","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.54","DNSServer2":"192.168.1.1","Mac":"84:F7:03:26:53:44","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
08:29:39.811 MQT: stat/tasmota_265344/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.54","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_265344","MqttUser":"saen","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
08:29:39.816 MQT: stat/tasmota_265344/STATUS7 = {"StatusTIM":{"UTC":"2022-12-06T07:29:39","Local":"2022-12-06T08:29:39","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"08:28","Sunset":"16:53"}}
08:29:39.822 MQT: stat/tasmota_265344/STATUS10 = {"StatusSNS":{"Time":"2022-12-06T08:29:39"}}
08:29:39.828 MQT: stat/tasmota_265344/STATUS11 = {"StatusSTS":{"Time":"2022-12-06T08:29:39","Uptime":"0T07:49:52","UptimeSec":28192,"Heap":179,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":45},"POWER":"OFF","Dimmer":98,"Color":"8EDEF9","HSBColor":"195,43,98","Channel":[56,87,98],"Scheme":4,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"MT-4EFD03","BSSId":"64:D1:54:4E:FD:04","Channel":9,"Mode":"11n","RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:03"}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
Just update from 12.1.0 > 12.2.0

### EXPECTED BEHAVIOUR
see values of internal temp sensor

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
using Home Assistant and after update, sensor is missing

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
It is not missing, it is a deliberate feature to not automatically display that mostly misleading number (not the actual internal temperature). If you still want it, you can enable it with `SetOption146`. Of course, if you had read the change log, you'd not have been surprised by this improvement :wink:

**Reply by Saentist** [reply]
> It is not missing, it is a deliberate feature to not automatically display that mostly misleading number (not the actual internal temperature). If you still want it, you can enable it with `SetOption146`. Of course, if you had read the change log, you'd not have been surprised by this improvement 😉

If I type in console 
```bash
SetOption146
```
result is 
```bash
11:51:18.204 CMD: SetOption146
11:51:18.208 MQT: stat/tasmota_265344/RESULT = {"SetOption146":"OFF"}
```
With i guess it turn off or it's turned off

----
p.s.
correct command must be 
```bash
SetOption146 ON
```
now I see Temp with is correct with offset of +25C

![image](https://user-images.githubusercontent.com/16975883/205894714-1c0dbf03-21c2-4a07-bdc8-33f2ba9674e6.png)

**Reply by sfromis** [reply]
You mean that you see a bullshit number, not anything "correct".

What I posted was what option is used to control it, not what you had to type to change it.

**Reply by Saentist** [reply]
It was always +25C from room temp.
Not sure where exactly sensor is.

**Reply by sfromis** [reply]
Don't count on it as room temperature indicator either, with different loads, the ESP32 may generate more or less heat.

**Reply by barbudor** [reply]
It's directly on the ESP processor die
So as sfromis said, may be impacted a lot by processor activity
